sdk/identity/azure-identity/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required (VERSION 3.12)
+
+set (AZURE_IDENTITY_HEADER
+    inc/azure/identity/client_secret_credential.hpp
+    inc/azure/identity/environment_credential.hpp
+)
+
+set (AZURE_IDENTITY_SOURCE
+    src/client_secret_credential.cpp
+    src/environment_credential.cpp
+)
+
+add_library(azure-identity ${AZURE_IDENTITY_HEADER} ${AZURE_IDENTITY_SOURCE})
+target_include_directories(azure-identity PUBLIC inc)
+target_link_libraries(azure-identity azure-core)
+
+# TODO: Do not read from version.txt, use version.hpp and replace this file
+# command with:
+# get_az_version("${CMAKE_CURRENT_SOURCE_DIR}/inc/azure/identity/<package-name>/version.hpp")
+# Access the variable from this function using ${AZ_LIBRARY_VERSION}
+# Look at the CMakeLists.txt for template package for example usage
+file(READ version.txt AZURE_IDENTITY_PACKAGE_VERSION)
```

Comment:
Fix before commiting this change.  There is code I wrote in C which you can use to grab the version in CMake.

`
if(EXISTS ${AZ_SDK_VERSION_H_FILE})
    file (STRINGS ${AZ_SDK_VERSION_H_FILE} VERSION_H_CONTENT)
    message(STATUS "Read version from ${AZ_SDK_VERSION_H_FILE}")

    #Find and read the AZ_SDK_VERSION_STRING define
    string(REGEX MATCHALL "#define AZ_SDK_VERSION_STRING \"([0-9]+)\.([0-9]+)\.([0-9]+)(\-[^\"\-]+)?\"" VERSION_PARTS ${VERSION_H_CONTENT})

    #Ensure we matched as expected.
    #   MAJOR.AZ_VERSION_MINOR.PATCH are required.
    #   PRERELEASE is optional.
    if(NOT CMAKE_MATCH_1 AND NOT CMAKE_MATCH_2 AND NOT CMAKE_MATCH_3)
        message(FATAL_ERROR "Unexpected version format in ${AZ_SDK_VERSION_H_FILE}")
    endif()
    set(AZ_SDK_VERSION_MAJOR ${CMAKE_MATCH_1} )
    set(AZ_SDK_VERSION_MINOR ${CMAKE_MATCH_2} )
    set(AZ_SDK_VERSION_PATCH ${CMAKE_MATCH_3} )
    if(CMAKE_MATCH_4)
        #Trim off the "-" from the start of pre-release portion
        # https://cmake.org/cmake/help/v3.10/command/string.html?highlight=substring#substring
        string(SUBSTRING ${CMAKE_MATCH_4} 1 -1 PRERELEASE_SUBSTRING )
        set(AZ_SDK_VERSION_PRERELEASE ${PRERELEASE_SUBSTRING})
    endif()
    message(STATUS "AZ_SDK_VERSION_MAJOR " ${AZ_SDK_VERSION_MAJOR})
    message(STATUS "AZ_SDK_VERSION_MAJOR " ${AZ_SDK_VERSION_MINOR})
    message(STATUS "AZ_SDK_VERSION_PATCH " ${AZ_SDK_VERSION_PATCH})
    message(STATUS "AZ_SDK_VERSION_PRERELEASE " ${AZ_SDK_VERSION_PRERELEASE})
else()
  message(FATAL_ERROR "Missing Version file ${AZ_SDK_VERSION_H_FILE}")
endif()
`

Reply:
This is what every other C++ SDK lib currently does. We need to do it, but it needs to be a separate PR.
I just opened an issue to track this - #750.

Reply:
Done for the identity module/this PR.

Reply:
I updated the issue I opened with the list of four Storage libraries still using the version.txt method:
* azure-storage-blobs
* azure-storage-common
* azure-storage-files-datalake
* azure-storage-files-shares